tests/v2/validation/upgrade/upgrade.go

```diff
@@ -0,0 +1,164 @@
+package upgrade
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/rancher/rancher/pkg/api/scheme"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	"github.com/rancher/rancher/tests/framework/extensions/ingresses"
+	"github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	corev1 "k8s.io/api/core/v1"
+	networkingv1 "k8s.io/api/networking/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	kubewait "k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apimachinery/pkg/watch"
+)
+
+// resourceNames struct contains the names of the resources
+type resourceNames struct {
+	core           map[string]string
+	coreWithSuffix map[string]string
+	random         map[string]string
+}
+
+var (
+	testContainer = &corev1.Container{
+		Name:  "test1",
+		Image: "ranchertest/mytestcontainer",
+	}
+	ingressHostName    = "sslip.io"
+	secretAsVolumeName = "secret-as-volume"
+)
+
+// waitUntilIngressIsAccessible waits until the ingress is accessible
+func waitUntilIngressIsAccessible(client *rancher.Client, hostname string) (bool, error) {
+	err := kubewait.Poll(500*time.Millisecond, 2*time.Minute, func() (done bool, err error) {
+		isIngressAccessible, err := ingresses.AccessIngressExternally(client, hostname, false)
+		if err != nil {
+			return false, err
+		}
+
+		return isIngressAccessible, nil
+	})
+
+	if err != nil && strings.Contains(err.Error(), kubewait.ErrWaitTimeout.Error()) {
+		return false, nil
+	}
+
+	return true, nil
+}
+
+// waitUntilIngressHostnameUpdates is a private function to wait until the ingress hostname updates
+func waitUntilIngressHostnameUpdates(client *rancher.Client, clusterID, namespace, ingressName string) error {
+	timeout := int64(60 * 5)
```

Comment:
Does this override the default timeout? Assuming 5 minutes (?) does the Update takes this much?

Reply:
Yes, this does override the default timeout. 	`WatchTimeoutSeconds = int64(60 * 30)`, the default timeout is 30 minutes, instead of the default one, it uses a 5 minutes timeout. The update usually takes around 1 - 2 mins, making this a bit longer to avoid connectivity issues.

Would it be better if it's around 3.5 minutes?

Reply:
I think giving it that extra time would be ok